internal/gitinterface/log_test.go

```diff
@@ -102,4 +108,171 @@ func TestGetCommitsBetweenRange(t *testing.T) {
+	commitIDs := make([]plumbing.Hash, 0, 6)
+
+	repo, err := git.Init(memory.NewStorage(), memfs.New())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	refName := plumbing.ReferenceName("refs/heads/main")
+
+	if err := repo.Storer.SetReference(plumbing.NewHashReference(refName, plumbing.ZeroHash)); err != nil {
+		t.Fatal(err)
+	}
+
+	ref, err := repo.Reference(refName, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	emptyBlobHash, err := WriteBlob(repo, []byte{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	treeHashes := make([]plumbing.Hash, 0, 6)
+	for i := 1; i <= 6; i++ {
+		objects := make([]object.TreeEntry, 0, i)
+		for j := 0; j < i; j++ {
+			objects = append(objects, object.TreeEntry{Name: fmt.Sprintf("%d", j+1), Hash: emptyBlobHash})
+		}
+
+		treeHash, err := WriteTree(repo, objects)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		treeHashes = append(treeHashes, treeHash)
+	}
+
+	// creating the first commit
+
+	commit := CreateCommitObject(testGitConfig, treeHashes[0], []plumbing.Hash{ref.Hash()}, fmt.Sprintf("Test commit %v", 1), testClock)
+
+	ref, err = applyCommitAndRefreshRef(repo, commit, ref)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	commitIDs = append(commitIDs, ref.Hash())
+
+	// creating two children from the first commit
+
+	children := make([]plumbing.Hash, 0, 2)
+	for i := 1; i <= 2; i++ {
+		commit := CreateCommitObject(testGitConfig, treeHashes[i], []plumbing.Hash{ref.Hash()}, fmt.Sprintf("Test commit %v", i+1), testClock)
+		childRefName := plumbing.ReferenceName(fmt.Sprintf("refs/heads/child%d", i))
+		childRef, err := applyCommitAndRefreshRef(repo, commit, plumbing.NewHashReference(childRefName, plumbing.ZeroHash))
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		commitIDs = append(commitIDs, childRef.Hash())
+		children = append(children, childRef.Hash())
+	}
+	// creating a merge commit from the two children
+
+	commit = CreateCommitObject(testGitConfig, treeHashes[4], children, fmt.Sprintf("Test commit %v", 5), testClock)
+
+	ref, err = applyCommitAndRefreshRef(repo, commit, ref)
+	if err != nil {
+		t.Fatal(err)
+	}
+	commitIDs = append(commitIDs, ref.Hash())
+
+	// creating children for each of the two children
+
+	commit = CreateCommitObject(testGitConfig, treeHashes[3], []plumbing.Hash{children[0]}, fmt.Sprintf("Test commit %v", 4), testClock)
+
+	ref, err = applyCommitAndRefreshRef(repo, commit, ref)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	commitIDs = append(commitIDs, ref.Hash())
+
+	commit = CreateCommitObject(testGitConfig, treeHashes[5], []plumbing.Hash{children[1]}, fmt.Sprintf("Test commit %v", 6), testClock)
+
+	ref, err = applyCommitAndRefreshRef(repo, commit, ref)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	commitIDs = append(commitIDs, ref.Hash())
+
+	// Git tree with merge commit structure without its commit trees and its values :
+	//
+	//  commit 4       commit 5         commit 6
+	//    │              │  │              │
+	//    └─► commit 2 ◄─┘  └─► commit 3 ◄─┘
+	// 		      │              │
```

Comment:
I think the spacing on this diagram is off for commit 1 to commit 2

Reply:
It's off in https://github.com/gittuf/gittuf/blob/94b5e2a50f068974d37bdd0a6e0ed8c03206e158/internal/gitinterface/log_test.go too though :\

The issue is there's a mix of `\t` and spaces. We can't control tab width, so I suggest using spaces throughout for these diagrams.

Reply:
fixed